kpm/lib/kpm/base_artifact.rb

```diff
@@ -164,8 +164,10 @@ def skip_if_exists(artifact_info, coordinates, sha1_file)
         end
 
         # Finally check if remote_sha1 matches what we have locally
-        local_sha1 = Digest::SHA1.file(artifact_info[:file_path]).hexdigest
-        local_sha1 == artifact_info[:sha1]
+        if !artifact_info[:is_tgz]
```

Comment:
Because the old code would always return false for tgz:
```ruby
if !File.exists?(artifact_info[:file_path]) || File.directory?(artifact_info[:file_path])
  return false
end
```